### ISSUE TITLE
Docker Build Failure: mistralrs-quant Fails with "No such file or directory" Error

### ISSUE THREAD

**Issue by ShivamSphn** [EricLBuehler/mistral.rs#893]
## Steps to Reproduce:

1. Run the Docker build command:
   ```bash
   docker build -f Dockerfile.cuda-all . -t mistral
   ```
2. Observe the error during the build process.

## Expected Behavior:
The build should complete successfully without errors.

## Actual Behavior:
The build fails with the following error:
```
thread 'main' panicked at mistralrs-quant/build.rs:15:22:
Failed to get compute cap: Os { code: 2, kind: NotFound, message: "No such file or directory" }
```

## Error
If the error is very long, please try to extract an excerpt of it and post a link to a [Gist](https://gist.github.com/).

## Other Information
- **Operating System:** Please specify (e.g., Windows, macOS, WSL2, Linux, etc.)
- **GPU or Accelerator Information:**
  - **CUDA Version:** 12.6
  - **Driver Version:** 560.35.03
  - **GPU Details:**
    ```plaintext
    NVIDIA-SMI 560.35.03              Driver Version: 560.35.03      CUDA Version: 12.6
    +-----------------------------------------------------------------------------------------+
    | GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
    | Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
    |                                         |                        |               MIG M. |
    |=========================================+========================+======================|
    |   0  NVIDIA A100 80GB PCIe          Off |   00000001:00:00.0 Off |                    0 |
    | N/A   38C    P0             54W /  300W |       1MiB /  81920MiB |      0%      Default |
    |                                         |                        |             Disabled |
    +-----------------------------------------+------------------------+----------------------+
    |   1  NVIDIA A100 80GB PCIe          Off |   00000002:00:00.0 Off |                    0 |
    | N/A   39C    P0             56W /  300W |       1MiB /  81920MiB |      0%      Default |
    |                                         |                        |             Disabled |
    +-----------------------------------------+------------------------+----------------------+
    | Processes:                                                                              |
    |  GPU   GI   CI        PID   Type   Process name                              GPU Memory |
    |        ID   ID                                                               Usage      |
    |=========================================================================================|
    |  No running processes found                                                             |
    +-----------------------------------------------------------------------------------------+
    ```

- **nvcc Version:**
  ```plaintext
  nvcc: NVIDIA (R) Cuda compiler driver
  Copyright (c) 2005-2023 NVIDIA Corporation
  Built on Fri_Jan__6_16:45:21_PST_2023
  Cuda compilation tools, release 12.0, V12.0.140
  Build cuda_12.0.r12.0/compiler.32267302_0
  ```

## Latest Commit or Version
v0.3.2.

**Additional Information:**

- **Dockerfile:** `Dockerfile.cuda-all`
- **Base Image:** `nvidia/cuda:12.4.1-cudnn-runtime-ubuntu22.04`
- **Rust Version:** Ensure the Rust version is compatible with the project requirements.
- **Environment Variables:**
  - `RUSTFLAGS="-Z threads=4"`
  - `RAYON_NUM_THREADS=4`
- **Relevant Code Section:**
  ```dockerfile
  RUN RUSTFLAGS="-Z threads=4" cargo build --release --workspace --exclude mistralrs-pyo3 --features "${FEATURES}"
  ```

**Comment by ShivamSphn** [comment]
Compiling gemm-f16 v0.17.1
   Compiling gemm v0.17.1
   Compiling candle-core v0.7.2 (https://github.com/EricLBuehler/candle.git?rev=629ec72#629ec72d)
error[E0063]: missing field `seed_value` in initializer of `cuda_backend::device::CudaDevice`
   --> /home/ubuntu/.cargo/git/checkouts/candle-c6a149c3b35a488f/629ec72/candle-core/src/cuda_backend/device.rs:203:12
    |
203 |         Ok(Self {
    |            ^^^^ missing `seed_value`

   Compiling hyper v1.5.0
   Compiling hyper-util v0.1.10
For more information about this error, try `rustc --explain E0063`.
error: could not compile `candle-core` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...



when i do cargo build --release --features "cuda flash-attn"

**Comment by EricLBuehler** [comment]
@ShivamSphn the build outside of docker should work now, I just pushed a fix.

It seems the `nvidia-smi` binary is missing from the docker container then. Can you reproduce the issue by running that in the container?

**Comment by ShivamSphn** [comment]
@EricLBuehler ```
83    Compiling anyhow v1.0.91
19.88    Compiling rustc-demangle v0.1.24
19.90    Compiling num-integer v0.1.46
19.92    Compiling tower-service v0.3.3
19.94    Compiling rayon v1.10.0
19.97    Compiling httparse v1.9.5
19.98    Compiling percent-encoding v2.3.1
19.98    Compiling stable_deref_trait v1.2.0
19.99 error: failed to run custom build command for `cudarc v0.12.1`
19.99
19.99 Caused by:
19.99   process didn't exit successfully: `/mistralrs/target/release/build/cudarc-460872b2cbd8fb22/build-script-build` (exit status: 101)
19.99   --- stdout
19.99   cargo:rerun-if-changed=build.rs
19.99   cargo:rerun-if-env-changed=CUDA_ROOT
19.99   cargo:rerun-if-env-changed=CUDA_PATH
19.99   cargo:rerun-if-env-changed=CUDA_TOOLKIT_ROOT_DIR
19.99
19.99   --- stderr
19.99   thread 'main' panicked at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/cudarc-0.12.1/build.rs:57:10:
19.99   Failed to execute `nvcc`: Os { code: 2, kind: NotFound, message: "No such file or directory" }
19.99   note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
19.99 warning: build failed, waiting for other jobs to finish...
------
Dockerfile:13
--------------------
  11 |
  12 |     # Build the project in release mode, excluding the specified workspace
  13 | >>> RUN cargo build --release  --features "cuda flash-attn" --workspace --exclude mistralrs-pyo3
  14 |
  15 |     # Stage 2: Minimal runtime environment
--------------------
ERROR: failed to solve: process "/bin/sh -c cargo build --release  --features \"cuda flash-attn\" --workspace --exclude mistralrs-pyo3" did not complete successfully: exit code: 101
➜  mistral.rs git:(master) ✗ ```



i used the latest dockerfile

**Comment by GaetanLepage** [comment]
@EricLBuehler would it be possible to alternatively provide the capability explicitly through an environment variable ?
This is what [candle does](https://github.com/huggingface/candle/blob/7ac0de15a9fafe59d9f97fb6d90662790488433e/candle-book/src/guide/installation.md?plain=1#L16) for instance.

**Comment by grahamking** [comment]
As a short term workaround, I put a shell script on the PATH in the container called `nvidia-smi`. It outputs what the real nvidia smi would output. It works!

Dockerfile:
```
COPY some-path/fake-nvidia-smi.sh /usr/local/cuda/bin/nvidia-smi
RUN chmod u+x /usr/local/cuda/bin/nvidia-smi
```

where that shell script is:
```
#!/bin/bash
echo "compute_cap"
echo "7.5"
```

**Comment by sammcj** [comment]
This looks like the same error as https://github.com/EricLBuehler/mistral.rs/issues/866

**Comment by EricLBuehler** [comment]
@ShivamSphn @sammcj this was fixed in #944 .
